saws/completer.py

```diff
@@ -137,10 +137,20 @@ def replace_substitution(self, text):
             A string representing input command text with a substitution,
             if one has been found.
         """
-        if '%s' in text:
+        text_to_find = '%s'
+
+        if text_to_find in text:
             tokens = text.split()
-            text = ' '.join(tokens[:-1])
-            text = re.sub('%s', tokens[-1], text)
+            text_to_find_index = self.text_utils.get_token_index(text_to_find,
+                                                                 tokens)
+            try:
+                token_after = tokens[text_to_find_index + 1]
+                # remove word immediately following %s
+                del tokens[text_to_find_index + 1]
+                text = ' '.join(tokens)
+                text = re.sub(text_to_find, token_after, text)
+            except:
+                return text
         return text
```

Comment:
@mlimaloureiro this is coming together nicely.  

I think it might be cleaner to use `pop()` instead of `del` since you want to element after removing to replace the `%s` marker. 

What do you think about making the following change?

```Python
substitution_marker = '%s'
if substitution_marker in text:
    tokens = text.split()
    replacement_index = self.text_utils.get_token_index(
        substitution_marker, tokens) + 1
    try:
        replacement_text = tokens.pop(replacement_index)
        text = ' '.join(tokens)
        text = re.sub(substitution_marker, replacement_text, text)
    except:
        return text
return text
```